openshift-knative-operator/pkg/common/api.go

```diff
@@ -0,0 +1,89 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	mf "github.com/manifestival/manifestival"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/discovery"
+)
+
+// UpgradePodDisruptionBudget upgrade the API version to policy/v1
+func UpgradePodDisruptionBudget() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "PodDisruptionBudget" {
+			return nil
+		}
+		if u.GetAPIVersion() != "policy/v1beta1" {
+			return nil
+		}
+		u.SetAPIVersion("policy/v1")
+		return nil
+	}
+}
+
+// UpgradeHorizontalPodAutoscaler upgrade the API version to autoscaling/v2
+func UpgradeHorizontalPodAutoscaler() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "HorizontalPodAutoscaler" {
+			return nil
+		}
+		if u.GetAPIVersion() != "autoscaling/v2beta2" {
+			return nil
+		}
+		u.SetAPIVersion("autoscaling/v2")
+		return nil
+	}
+}
+
+// CheckMinimumVersion checks if the version in the arg meets the requirement or not.
```

Comment:
the requirement is a little hard to understand.
Can we perhaps link to its definition?

Reply:
Yeah will do it is a relic of the past.